metadata-io/src/main/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphService.java

```diff
@@ -114,31 +115,51 @@ public RelatedEntitiesResult findRelatedEntities(
             offset, count)
     );
 
-    final String srcCriteria = filterToCriteria(sourceEntityFilter);
-    final String destCriteria = filterToCriteria(destinationEntityFilter);
+    final String srcCriteria = filterToCriteria(sourceEntityFilter).trim();
+    final String destCriteria = filterToCriteria(destinationEntityFilter).trim();
     final String edgeCriteria = relationshipFilterToCriteria(relationshipFilter);
 
     final RelationshipDirection relationshipDirection = relationshipFilter.getDirection();
 
-    String matchTemplate = "MATCH (src%s %s)-[r%s %s]-(dest%s %s)";
+    String matchTemplate = "MATCH (src %s)-[r%s %s]-(dest %s)%s";
     if (relationshipDirection == RelationshipDirection.INCOMING) {
-      matchTemplate = "MATCH (src%s %s)<-[r%s %s]-(dest%s %s)";
+      matchTemplate = "MATCH (src %s)<-[r%s %s]-(dest %s)%s";
     } else if (relationshipDirection == RelationshipDirection.OUTGOING) {
-      matchTemplate = "MATCH (src%s %s)-[r%s %s]->(dest%s %s)";
+      matchTemplate = "MATCH (src %s)-[r%s %s]->(dest %s)%s";
     }
 
-    final String returnNodes = String.format("RETURN dest%s, type(r)", destinationType); // Return both related entity and the relationship type.
+    final String returnNodes = String.format("RETURN dest, type(r)"); // Return both related entity and the relationship type.
     final String returnCount = "RETURN count(*)"; // For getting the total results.
 
     String relationshipTypeFilter = "";
     if (relationshipTypes.size() > 0) {
       relationshipTypeFilter = ":" + StringUtils.join(relationshipTypes, "|");
     }
 
-    // Build Statement strings
+    String whereClause = "";
+
+    Boolean hasSourceTypes = sourceTypes != null && !sourceTypes.isEmpty();
```

Comment:
any chance we can break this longer block into a sep method? 

maybe

computeEntityTypeWhereClause

Reply:
done